splinepy/io/cats.py

```diff
@@ -0,0 +1,306 @@
+            if CATS_XML_KEY_WORDS["coeff_names"] in info.tag:
+                debug(
+                    f"Will use the following control point vars as control "
+                    f"point coordinates : {info.text.split()[:dim]}"
+                )
+            # Control points
+            elif CATS_XML_KEY_WORDS["control_points"] in info.tag:
+                spline_dict["control_points"] = np.fromstring(
+                    info.text.replace("\n", " "), sep=" "
+                ).reshape(-1, ctps_dim)[:, :dim]
+
+            # degrees
+            elif CATS_XML_KEY_WORDS["degrees"] in info.tag:
+                spline_dict["degrees"] = np.fromstring(
+                    info.text.replace("\n", " "), sep=" "
+                )
+                if spline_dict["degrees"].size != para_dim:
+                    raise ValueError(
+                        f"Conflicting information provided. Got "
+                        f"{spline_dict['degrees'].size} degrees, but "
+                        f"parametric dimension is {para_dim}"
+                    )
+
+            # weights
+            elif CATS_XML_KEY_WORDS["weights"] in info.tag:
+                spline_dict["weights"] = np.fromstring(
+                    info.text.replace("\n", " "), sep=" "
+                )
+
+            # knot vectors
+            elif CATS_XML_KEY_WORDS["knot_vectors"] in info.tag:
+                spline_dict["knot_vectors"] = []
+                for child_info in info:
+                    if not CATS_XML_KEY_WORDS["knot_vector"] in child_info.tag:
+                        debug("Redundant item in knot_vectors block of xml")
+                    spline_dict["knot_vectors"].append(
+                        np.fromstring(
+                            child_info.text.replace("\n", " "), sep=" "
+                        )
+                    )
+
+            # All other keywords will be ignored for the moment
+            else:
+                debug(f"Ignoring keyword {info.tag}")
+
+        return spline_dict
+
+    # Parse  XML file
+    debug(f"Parsing xml-file '{fname}' ...")
+    root = ET.parse(fname).getroot()
+    debug("XML-file parsed start conversion")
+    list_of_splines = []
+    for child in root:
+        if child.tag.startswith(CATS_XML_KEY_WORDS["spline_list"]):
+            if child.attrib.get(CATS_XML_KEY_WORDS["spline_type"], "1") != "1":
+                raise ValueError(
+                    f"Unknown SplineType "
+                    f"{child.attrib[CATS_XML_KEY_WORDS['spline_type']]}"
+                )
+            n_patches = int(
+                child.attrib.get(CATS_XML_KEY_WORDS["n_patches"], 0)
+            )
+            for patch_element in child:
+                list_of_splines.append(_read_spline(patch_element))
+        else:
+            debug(f"Unused xml-keyword {child.tag}")
+    debug(f"Found a total of {len(list_of_splines)} " f"BSplines and NURBS")
+    if len(list_of_splines) != n_patches:
```

Comment:
```
    if len(list_of_splines) != n_patches:
UnboundLocalError: local variable 'n_patches' referenced before assignment
```
I guess it doesn't reach l.152 ?

Reply:
Previously root was "xml" however, it is now updated to `SplineList`